### PR TITLE
Add support for .NET 4.x Scripting Runtime Version

### PR DESCRIPTION
**Pull request by VoxelBoy**
unity-shell wasn't working when imported into Unity projects using .NET 4.x Scripting Runtime Version (in ProjectSettings); the Evaluator could not be constructed.

I've added a 4.x variant of the Mono.CSharp.dll from Unity.app/Contents/MonoBleedingEdge/lib/mono/4.5/. I've also added "#if NET_4_6 || NET_STANDARD_2_0" directives to the csharp scripts so they work with both runtimes.

I had the problem that both Mono.CSharp.dlls (one for 3.5 and one for 4.x) could not be enabled for the Editor platform at the same time. So I left the 3.5 one enabled by default. To get the package to work with 4.x runtime, Mono.CSharp.4.x.dll needs to be enabled for Editor platform and Mono.CSharp.3.5.dll needs to be disabled.

### CODE REVIEW

File: Assets/UnityShell/Editor/Scripts/AutocompleteProvider.cs [hunk 1]
```diff
@@ -18,19 +14,28 @@ public class AutocompleteProvider
...
-			new Thread(() =>
+			new Thread((threadStart) =>
 			{
 				int handle = handleCount;
 
 				if(!string.IsNullOrEmpty(input))
 				{
 					string prefix;
+#if NET_4_6 || NET_STANDARD_2_0
+					var evaluator = new Evaluator(new CompilerContext(new CompilerSettings(), new ConsoleReportPrinter()));
```
Comment (marijnz) [inline Assets/UnityShell/Editor/Scripts/AutocompleteProvider.cs]: As this is a new Evaluator, does Autocomplete still work exactly the same, as usings that are in ShellEveluator are not applied to this evaluator?

Comment (marijnz) [reply Assets/UnityShell/Editor/Scripts/AutocompleteProvider.cs]: @VoxelBoy ?

Comment (VoxelBoy) [reply Assets/UnityShell/Editor/Scripts/AutocompleteProvider.cs]: Haven't had the time, sorry. I'll try looking into this today.

Comment (VoxelBoy) [reply Assets/UnityShell/Editor/Scripts/AutocompleteProvider.cs]: Fixed. In my testing, autocomplete now comes up faster for 4.x runtime than 3.5.

**Comment by marijnz** [comment]
Great, good to be merged! 👍

**Pull request merged**